apps/condo/domains/ticket/components/TicketClassifierSelect.tsx

```diff
@@ -0,0 +1,320 @@
+    initialValues: {
+        classifierRule: string
+        placeClassifier: string
+        categoryClassifier: string
+        descriptionClassifier?: string
+    }
+}
+interface ITicketThreeLevelsClassifierHookOutput {
+    ClassifiersEditorComponent: React.FC<{ form, disabled }>
+}
+
+interface ITicketClassifierSelectHookInput {
+    onChange: (id: string) => void
+    onSearch: (id: string) => void
+    initialValue: string | null
+}
+
+type ClassifierSelectComponent = React.FC<{ disabled?: boolean, style?: React.CSSProperties, value?: string }>
+interface ITicketClassifierSelectHookOutput {
+    set: {
+        all: React.Dispatch<React.SetStateAction<Options[]>>
+        one: React.Dispatch<React.SetStateAction<string>>
+        search: React.Dispatch<React.SetStateAction<Options[]>>
+    }
+    SelectComponent: ClassifierSelectComponent
+    ref: React.MutableRefObject<HTMLSelectElement>
+}
+
+const useTicketClassifierSelectHook = ({
+    onChange,
+    onSearch,
+    initialValue,
+}: ITicketClassifierSelectHookInput): ITicketClassifierSelectHookOutput => {
+    const [selected, setSelected] = useState<string>(null)
+    const [classifiers, setClassifiersFromRules] = useState<Options[]>([])
+    const [searchClassifiers, setSearchClassifiers] = useState<Options[]>([])
+    const classifiersRef = useRef<HTMLSelectElement>(null)
+    const optionsRef = useRef<Options[]>([])
+
+    const setClassifiers = (classifiers) => {
+        setClassifiersFromRules(classifiers)
+        // We need to remove search classifiers when rules start to work
+        setSearchClassifiers([])
+    }
+
+    useEffect(() => {
+        optionsRef.current = uniqBy([...classifiers, ...searchClassifiers], 'id')
+    }, [classifiers, searchClassifiers])
+
+    const SelectComponent = useMemo(() => {
+        const SelectComponentWrapper: ClassifierSelectComponent = (props) => {
+            const { disabled, style } = props
+            return (
+                <Select
+                    showSearch
+                    style={style}
+                    allowClear={true}
+                    onSelect={onChange}
+                    onSearch={onSearch}
+                    onClear={() => onChange(null)}
+                    optionFilterProp={'title'}
+                    defaultActiveFirstOption={false}
+                    disabled={disabled}
+                    defaultValue={initialValue}
+                    ref={classifiersRef}
+                    showAction={['focus', 'click']}
+                >
+                    {
+                        optionsRef.current.map(classifier => (
+                            <Option value={classifier.id} key={classifier.id} title={classifier.name}>{classifier.name}</Option>
+                        ))
+                    }
+
+                </Select>
+            )
+        }
+        return SelectComponentWrapper
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
remove eslint disable, add missed deps or rewrite hook logic

Reply:
This hook needs to be run only once on create - so it has empty deps

Reply:
Done